### DISCUSSION TITLE
How to rewrite the search URL?

### DISCUSSION THREAD

**Discussion by ghost** [searx/searx#2510] [Installation issue: custom]
Hi Team,

I want to rewrite the search URL as directory structure automatically after search, please help.
Existing URL: https://domain-name/?q=search+query
Looking solution for: https://domain-name/q/search-query

**Comment by unixfox** [comment]
It depends on which web server you are using. Is it NGINX or something else?

**Comment by ghost** [comment]
> It depends on which web server you are using. Is it NGINX or something else?

Hi unixfox,
thanks for the reply.
i am using it on apache.. apache has been configured as reverse proxy.

**Comment by ghost** [comment]
Contents of my Apache Conf file as below (real domain name masked here with fake)

<VirtualHost *:80>
ServerAdmin webmaster@localhost
DocumentRoot /var/www/html
ErrorLog ${APACHE_LOG_DIR}/error.log
CustomLog ${APACHE_LOG_DIR}/access.log combined

ProxyPreserveHost On
ProxyPass "/" "http://127.0.0.1:8888/"
ProxyPassReverse "/" "http://127.0.0.1:8888/"
ServerName FAKE-DOMAIN-NAME.com

</VirtualHost>

**Comment by unixfox** [comment]
I don't often use Apache enough for helping you about that but here are some links that will guide you in order to achieve what you want:
- https://stackoverflow.com/questions/13003319/htaccess-rewrite-query-string-as-path
- https://www.taniarascia.com/rewrite-query-string-to-path-with-htaccess/
- https://perishablepress.com/redirect-query-string-htaccess/

I think if you really need some help about that it's better to ask on a forum dedicated for apache web server because what you want is unrelated to the Searx code itself and hardly anybody uses Apache in the frequent visitors on the Searx github repository.

**Comment by ghost** [comment]
Thanks for your quick help.
From apache side I am aware of how to create URL rewrites in .htaccess
But in this case if I create the rules in .htaccess, still the searx page keeps generating /?q=xxxx URLs; so this is not serving the purpose.
So I want your help to understand how can I completely change the URL structure from  /?q=xxxx  to  /q/xxxxx from searx core files.

**Comment by ghost** [comment]
Hi,

Can you please help to understand from which file I can edit below
parameters?

{{ suggestion.url }}  and {{ q }}
